### PR TITLE
intial setup for plugin for Okta Integration

### PR DESCRIPTION
**Pull request by codetheorem**
Signed-off-by: Hrishikesh Agarwal <hrishikeshagarwalv@gmail.com>

Added initial files needed for the development of the Okta Plugin for the Strapi.

### CODE REVIEW

**Comment by codetheorem** [comment]
screencapture-localhost-8000-admin-plugins-my-plugin-1657920998298

UI of the new Plugin.

File: package.json [hunk 1]
```diff
@@ -11,18 +11,21 @@
...
-    "strapi-plugin-documentation": "^3.0.1",
+    "strapi-plugin-documentation": "3.0.1",
     "strapi-plugin-email": "3.0.1",
-    "strapi-plugin-graphql": "^3.0.1",
+    "strapi-plugin-graphql": "3.0.1",
     "strapi-plugin-upload": "3.0.1",
     "strapi-plugin-users-permissions": "3.0.1",
-    "strapi-provider-email-sendgrid": "^3.0.2",
-    "strapi-provider-upload-cloudinary": "^3.0.4",
+    "strapi-provider-email-sendgrid": "3.0.2",
```
Comment (cyntss) [inline package.json:27]: why was the `^` removed from all these lines?

File: plugins/okta/admin/src/components/InputMedia/index.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            type={getInputType(type)}
+            validations={validations}
+            value={inputValue}
+            withDefaultValue={false}
+          />
+        );
+      }}
+    </FormattedMessage>
+  );
+}
```
Comment (cyntss) [inline plugins/okta/admin/src/components/InputMedia/index.js]: need to add a new line at the end of the file

File: plugins/okta/admin/src/containers/HomePage/index.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
+/*
+ *
+ * HomePage
+ *
+ */
+
+import React, { memo } from "react";
+// import PropTypes from 'prop-types';
```
Comment (cyntss) [inline plugins/okta/admin/src/containers/HomePage/index.js]: let's remove this comment

File: plugins/okta/admin/src/containers/HomePage/index.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+import pluginId from "../../pluginId";
+import { Label, InputText } from "@buffetjs/core";
+import okta_logo from "../../assets/okta_logo.png";
+
+const HomePage = () => {
+  const [val, setValue] = React.useState("");
+
+  return (
+    <div style={{ paddingLeft: "10%", paddingRight: "10%" }}>
+      <h1>
```
Comment (cyntss) [inline plugins/okta/admin/src/containers/HomePage/index.js]: let's remove the h1. The image doesn't need to be inside h1

File: plugins/okta/admin/src/index.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    lifecycles,
+    leftMenuLinks: [],
+    leftMenuSections: [],
+    mainComponent: App,
+    name: pluginPkg.strapi.name,
+    preventComponentRendering: false,
+    trads,
+  };
+
+  // strapi.registerField({ type: 'media', Component: InputMedia });
```
Comment (cyntss) [inline plugins/okta/admin/src/index.js]: let's remove this comment

File: plugins/okta/admin/src/index.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import pluginPkg from '../../package.json';
+import pluginId from './pluginId';
+import App from './containers/App';
+import Initializer from './containers/Initializer';
+import lifecycles from './lifecycles';
+import trads from './translations';
+import InputMedia from './components/InputMedia';
+
+
+export default strapi => {
```
Comment (cyntss) [inline plugins/okta/admin/src/index.js]: I'm not sure why we're calling it strapi. Should we name this method "Okta" ?

File: plugins/okta/admin/src/lifecycles.js [hunk 7]
```diff
@@ -0,0 +1,3 @@
+function lifecycles() {}
```
Comment (cyntss) [inline plugins/okta/admin/src/lifecycles.js]: Is this needed?

File: plugins/okta/admin/src/utils/getTrad.js [hunk 8]
```diff
@@ -0,0 +1,5 @@
+import pluginId from '../pluginId';
+
+const getTrad = id => `${pluginId}.${id}`;
```
Comment (cyntss) [inline plugins/okta/admin/src/utils/getTrad.js]: Can we rename it to getL10n ? since `trad` isn't a good name for translations. 
Also update it in file name and other files where we use "Trad" as name

File: plugins/okta/config/routes.json [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+    {
+      "method": "GET",
+      "path": "/",
+      "handler": "okta.index",
+      "config": {
+        "policies": []
+      }
+    },
+    {
+      "method": "GET",
```
Comment (cyntss) [inline plugins/okta/config/routes.json]: the two objects seem identical. Is there a reason for this?

File: plugins/okta/package.json [hunk 10]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "strapi-plugin-my-plugin",
```
Comment (cyntss) [inline plugins/okta/package.json]: let's rename it to okta-strapi-plugin

File: plugins/okta/package.json [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+  "version": "0.0.0",
+  "description": "This is the description of the plugin.",
```
Comment (cyntss) [inline plugins/okta/package.json]: "description": "Integrate SSO with Okta on Strapi with this plugin."

File: plugins/okta/package.json [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+  },
+  "dependencies": {},
+  "author": {
+    "name": "A Strapi developer",
+    "email": "",
+    "url": ""
+  },
+  "maintainers": [
+    {
+      "name": "A Strapi developer",
```
Comment (cyntss) [inline plugins/okta/package.json]: EOS UX/UI Solutions

File: plugins/okta/package.json [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+      "email": "",
+      "url": ""
```
Comment (cyntss) [inline plugins/okta/package.json]: userstory.site

File: plugins/okta/services/okta.js [hunk 14]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+/**
+ * okta.js service
+ *
+ * @description: A set of functions similar to controller's actions to avoid code duplication.
+ */
+
+module.exports = {
```
Comment (cyntss) [inline plugins/okta/services/okta.js]: we're not exporting anything. is this file needed?

**Review by cyntss — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (codetheorem) [reply package.json:27]: Some packages were not compatible with the new patches and were throwing errors.

File: plugins/okta/services/okta.js (hunk 14 above)
Comment (codetheorem) [reply plugins/okta/services/okta.js]: Not needed currently but comes as a part of boilerplate.

File: plugins/okta/admin/src/lifecycles.js (hunk 7 above)
Comment (codetheorem) [reply plugins/okta/admin/src/lifecycles.js]: Not needed currently but comes as a part of boilerplate.

File: plugins/okta/services/okta.js (hunk 14 above)
Comment (cyntss) [reply plugins/okta/services/okta.js]: let's remove everything that's not needed to reduce clutter

File: plugins/okta/admin/src/lifecycles.js (hunk 7 above)
Comment (cyntss) [reply plugins/okta/admin/src/lifecycles.js]: let's remove everything that's not needed to reduce clutter

**Review by cyntss — CHANGES_REQUESTED** [review]

**Review by cyntss — APPROVED** [review]
